bofire/strategies/stepwise/conditions.py

```diff
@@ -0,0 +1,53 @@
+from abc import abstractmethod
+from typing import Union
+
+import pandas as pd
+
+import bofire.data_models.strategies.stepwise.conditions as data_models
+from bofire.data_models.domain.api import Domain
+
+
+class Condition(object):
+    @abstractmethod
+    def evaluate(self, domain: Domain, experiments: pd.DataFrame) -> bool:
+        pass
+
+
+class RequiredExperimentsCondition(Condition):
+    def __init__(self, data_model: data_models.RequiredExperimentsCondition):
+        self.n_required_experiments = data_model.n_required_experiments
+
+    def evaluate(self, domain: Domain, experiments: pd.DataFrame) -> bool:
+        n_experiments = len(
+            domain.outputs.preprocess_experiments_all_valid_outputs(experiments)
+        )
+        return n_experiments >= self.n_required_experiments
+
+
+class CombiCondition(Condition):
+    def __init__(self, data_model: data_models.CombiCondition) -> None:
+        self.conditions = [map(c) for c in data_model.conditions]  # type: ignore
+        self.n_required_conditions = data_model.n_required_conditions
+
+    def evaluate(self, domain: Domain, experiments: pd.DataFrame) -> bool:
+        n_matched_conditions = 0
+        for c in self.conditions:
```

Comment:
I'd expect a combination of conditions to combine them rather with `or`  or `and`. Not sure what the use case of counting the number `True`s could be.

Reply:
Yep, but I was lazy and wanted to have it in one class. You can have `and` via a list of two conditions and `n_required_conditions=2`  and `or` via also two conditons and `n_required_conditions=1`. 

Of course, we can also replace the `n_required_conditions` field by an `operator` field which takes a literal of being either `"and"` or `"or"`. But this would be less expressive as my solution.

What do you think?

Reply:
Ah. Ok. Got it.